lib/fog/aliyun/models/compute/zones.rb

```diff
@@ -0,0 +1,14 @@
+module Fog
+  module Compute
+    class Aliyun
+      class Zones < Fog::Collection
+        model Fog::Compute::Aliyun::Zone
+
+        def all
+          data = Fog::JSON.decode(service.list_zones.body)['Zones']['Zone']
```

Comment:
You should have a request to do the coding and decoding part of the process.
Could you try to use one instead?

Reply:
If I see rest of the model implementation then decode method has been used. Do you mean there should be a common request to handle coding and decoding part for all models ? if not can you please elaborate more.